# Fair warning: Do NOT run this without a VPN. The GPU chip shortage + out-of-control scalper situation made many tech websites hyper-vigilant of bots scraping for data to buy up the cards between 2020 and 2022.

### Running this without a VPN will likely get you IP banned from every website listed.

# The goal of this project is to scrape for the best deals on Graphics cards across the internet, both New and Used (ebay). It is also a way to see at any point of time what cards give you the best bang for the buck.

In [1]:
import requests
from bs4 import BeautifulSoup

In [2]:
link='https://www.videocardbenchmark.net/high_end_gpus.html'

In [3]:
html=requests.get(link).text

In [4]:
soup = BeautifulSoup(html, "lxml")

In [5]:
gpunames=list(soup.findAll('span',{'class':'prdname'}))
gpulist=[]
for entry in gpunames:
    new=str(entry)
    new=str.replace(new,'<span class="prdname">','')
    new=str.replace(new,'</span>','',)
    gpulist.append(new)
len(gpulist)

1154

In [6]:
newprices=list(soup.findAll('span',{'class':"price-neww"}))
newpricelist=[]
for entry in newprices:
    new=str(entry)
    new=str.replace(new,'<span class="price-neww">','')
    new=str.replace(new,'</span>','')
    new=str.replace(new,'*','')
    new=str.replace(new,'NA','np.nan')
    newpricelist.append(new)
len(newpricelist)

#checking to make sure the size of both lists are equal

1154

In [7]:
scores=list(soup.findAll('span',{'class' :'count'}))
scorelist=[]
for entry in scores:
    new=str(entry)
    new=str.replace(new,'<span class="count">','')
    new=str.replace(new,'</span>','')
    scorelist.append(new)
len(scorelist)

#checking to make sure the size of all lists are equal

1154

In [8]:
import pandas as pd
pd.set_option("display.max_columns",None)
pd.set_option("display.max_rows",None)
import numpy as np
df=pd.DataFrame([gpulist,newpricelist,scorelist]).T

df=df.reset_index()
df.rename(columns={'index':'RANKING',0:'GPU',1:'NEW_PRICE',2:'PASSMARK_SCORE'},inplace=True)
df['RANKING']=df['RANKING']+1
df=df.head(158) #removing irrelevant older GPUs
                #worst gpu in the list is the RX 580, for those interested.

In [9]:
usedprice=[1274.95,909.99,862.49,900.00,693.98,633.90,4042.44,
 599.95,1499,549.99,469.95,459.95,np.nan,np.nan,524.99,
 440,np.nan,800,799,np.nan,349.99,371.70,
 np.nan,500,np.nan,np.nan,399.97,349.95,np.nan,
 np.nan,np.nan,np.nan,249.99,np.nan,315,np.nan,
 np.nan,345,np.nan,np.nan,312.25,np.nan,1500,
 650,np.nan,np.nan,277.77,np.nan,np.nan,293.25,
 np.nan,np.nan,np.nan,np.nan,349.95,292.00,np.nan,619.99,
 np.nan,201.75,np.nan,np.nan,np.nan,np.nan,np.nan,
 np.nan,274.99,229.99,np.nan,189.97,np.nan,np.nan,
 240,215,np.nan,162.45,np.nan,210.70,np.nan,219,234.77,
 168,219,np.nan,np.nan,np.nan,np.nan,np.nan,
 np.nan,np.nan,170,300,np.nan,np.nan,np.nan,np.nan,np.nan,
 np.nan,np.nan,np.nan,np.nan,195,np.nan,140,np.nan,
 np.nan,np.nan,np.nan,np.nan,125,np.nan,np.nan,210.70,
 np.nan,np.nan,np.nan,np.nan,300,np.nan,np.nan,np.nan,
 np.nan,np.nan,np.nan,np.nan,np.nan,160,np.nan,np.nan,
 132.45,153.26,np.nan,np.nan,np.nan,np.nan,91.75,np.nan,np.nan,
 76.20,159.99,120,160,np.nan,np.nan,np.nan,87,
 np.nan,np.nan,np.nan,150,np.nan,np.nan,np.nan,np.nan,
 78,np.nan,np.nan,120
]
df['USED_PRICE']=usedprice

df

RANKING                                 GPU NEW_PRICE PASSMARK_SCORE  \
0          1                 GeForce RTX 3090 Ti  1,399.94         30,187   
1          2                   Radeon RX 6950 XT  1,256.20         27,987   
2          3                 GeForce RTX 3080 Ti    899.99         27,207   
3          4                    GeForce RTX 3090  1,203.18         26,560   
4          5                   Radeon RX 6900 XT    729.99         25,757   
5          6                    GeForce RTX 3080    759.99         25,132   
6          7                           RTX A5500    np.nan         23,978   
7          8                   Radeon RX 6800 XT    669.99         23,793   
8          9                           RTX A5000  2,631.20         23,628   
9         10                 GeForce RTX 3070 Ti    669.99         23,572   
10        11                    GeForce RTX 3070    549.99         22,209   
11        12                 GeForce RTX 2080 Ti    998.59         21,817   
12        13                           RTX A6000  4,999.99         21,770   
13        14                           RTX A4500  1,752.60         21,704   
14        15                      Radeon RX 6800    579.99         20,868   
15        16                 GeForce RTX 3060 Ti    469.99         20,359   
16        17      GeForce RTX 3080 Ti Laptop GPU    np.nan         20,173   
17        18                           TITAN RTX  2,794.99         20,051   
18        19                             TITAN V  1,999.99         19,773   
19        20                     Quadro RTX 6000  6,300.00         19,566   
20        21              GeForce RTX 2080 SUPER    683.99         19,509   
21        22                            TITAN Xp    np.nan         19,327   
22        23                    Radeon PRO W6800  2,194.99         19,283   
23        24                   Radeon RX 6750 XT    529.99         19,205   
24        25                     Quadro RTX 8000  4,741.49         19,126   
25        26                           RTX A4000  1,233.98         18,895   
26        27                   Radeon RX 6700 XT    449.99         18,887   
27        28                    GeForce RTX 2080    633.03         18,727   
28        29                      Radeon RX 6700    np.nan         18,726   
29        30                                A10G    np.nan         18,724   
30        31         TITAN Xp COLLECTORS EDITION    np.nan         18,711   
31        32      GeForce RTX 3070 Ti Laptop GPU    np.nan         18,539   
32        33                 GeForce GTX 1080 Ti    483.82         18,361   
33        34                     NVIDIA TITAN Xp  1,645.00         18,280   
34        35              GeForce RTX 2070 SUPER    499.99         18,188   
35        36                RTX A4500 Laptop GPU    np.nan         17,731   
36        37                        Quadro GV100  8,999.00         17,675   
37        38                    GeForce RTX 3060    329.00         17,006   
38        39                 TITAN V CEO Edition    np.nan         16,988   
39        40                RTX A5500 Laptop GPU    np.nan         16,981   
40        41                   Radeon RX 5700 XT    694.94         16,899   
41        42                           Tesla P40    np.nan         16,864   
42        43                      Radeon Pro VII    np.nan         16,853   
43        44                          Radeon VII  1,899.95         16,804   
44        45  Radeon RX 5700 XT 50th Anniversary    np.nan         16,644   
45        46         GeForce RTX 3080 Laptop GPU    np.nan         16,575   
46        47              GeForce RTX 2060 SUPER    359.99         16,508   
47        48                        Quadro GP100    np.nan         16,203   
48        49                     Quadro RTX 5000    np.nan         16,163   
49        50                    GeForce RTX 2070    627.44         16,083   
50        51                RTX A5000 Laptop GPU    np.nan         16,044   
5

In [10]:
list_=[]
for entry in df['NEW_PRICE']:
    list_.append(str.replace(entry,',',''))
df['NEW_PRICE']=list_
df['NEW_PRICE']=df['NEW_PRICE'].replace('np.nan',0)
df['NEW_PRICE']=df['NEW_PRICE'].astype('float64')
df['NEW_PRICE']=df['NEW_PRICE'].replace(0,np.nan)

list_=[]
for entry in df['PASSMARK_SCORE']:
    list_.append(str.replace(entry,',',''))
df['PASSMARK_SCORE']=list_
df['PASSMARK_SCORE']=df['PASSMARK_SCORE'].astype('float64')
del list_
df

RANKING                                 GPU  NEW_PRICE  PASSMARK_SCORE  \
0          1                 GeForce RTX 3090 Ti    1399.94         30187.0   
1          2                   Radeon RX 6950 XT    1256.20         27987.0   
2          3                 GeForce RTX 3080 Ti     899.99         27207.0   
3          4                    GeForce RTX 3090    1203.18         26560.0   
4          5                   Radeon RX 6900 XT     729.99         25757.0   
5          6                    GeForce RTX 3080     759.99         25132.0   
6          7                           RTX A5500        NaN         23978.0   
7          8                   Radeon RX 6800 XT     669.99         23793.0   
8          9                           RTX A5000    2631.20         23628.0   
9         10                 GeForce RTX 3070 Ti     669.99         23572.0   
10        11                    GeForce RTX 3070     549.99         22209.0   
11        12                 GeForce RTX 2080 Ti     998.59         21817.0   
12        13                           RTX A6000    4999.99         21770.0   
13        14                           RTX A4500    1752.60         21704.0   
14        15                      Radeon RX 6800     579.99         20868.0   
15        16                 GeForce RTX 3060 Ti     469.99         20359.0   
16        17      GeForce RTX 3080 Ti Laptop GPU        NaN         20173.0   
17        18                           TITAN RTX    2794.99         20051.0   
18        19                             TITAN V    1999.99         19773.0   
19        20                     Quadro RTX 6000    6300.00         19566.0   
20        21              GeForce RTX 2080 SUPER     683.99         19509.0   
21        22                            TITAN Xp        NaN         19327.0   
22        23                    Radeon PRO W6800    2194.99         19283.0   
23        24                   Radeon RX 6750 XT     529.99         19205.0   
24        25                     Quadro RTX 8000    4741.49         19126.0   
25        26                           RTX A4000    1233.98         18895.0   
26        27                   Radeon RX 6700 XT     449.99         18887.0   
27        28                    GeForce RTX 2080     633.03         18727.0   
28        29                      Radeon RX 6700        NaN         18726.0   
29        30                                A10G        NaN         18724.0   
30        31         TITAN Xp COLLECTORS EDITION        NaN         18711.0   
31        32      GeForce RTX 3070 Ti Laptop GPU        NaN         18539.0   
32        33                 GeForce GTX 1080 Ti     483.82         18361.0   
33        34                     NVIDIA TITAN Xp    1645.00         18280.0   
34        35              GeForce RTX 2070 SUPER     499.99         18188.0   
35        36                RTX A4500 Laptop GPU        NaN         17731.0   
36        37                        Quadro GV100    8999.00         17675.0   
37        38                    GeForce RTX 3060     329.00         17006.0   
38        39                 TITAN V CEO Edition        NaN         16988.0   
39        40                RTX A5500 Laptop GPU        NaN         16981.0   
40        41                   Radeon RX 5700 XT     694.94         16899.0   
41        42                           Tesla P40        NaN         16864.0   
42        43                      Radeon Pro VII        NaN         16853.0   
43        44                          Radeon VII    1899.95         16804.0   
44        45  Radeon RX 5700 XT 50th Anniversary        NaN         16644.0   
45        46         GeForce RTX 3080 Laptop GPU        NaN         16575.0   
46        47              GeForce RTX 2060 SUPER     359.99         16508.0   
47        48                        Quadro GP100        NaN         16203.0   
48        49                     Quadro RTX 5000        NaN         16163.0   
49        50                    GeForce RTX 2070     62

NVS 315 has an invalid score (1gb vram card, there's no way it outperforms a GTX 970 which has 4gb).

Many new prices are incorrect.  
I will be removing many New Prices as many are both incorrect and for older cards. The 'new' prices for many older cards will be 2-3x MSRP, making their value proposition completely unreasonable.

Fix the scores. 

In [11]:
import re

list_of_scores_to_fix = [['GeForce RTX 3090 Ti',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3090ti/'],
                         ['Radeon RX 6950 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6950xt/'],
                         ['GeForce RTX 3080 Ti',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3080ti/'],
                         ['GeForce RTX 3090',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3090/'],
                         ['Radeon RX 6900 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6900xt/'],
                         ['GeForce RTX 3080',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3080/'],
                         ['Radeon RX 6800 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6800xt/'],
                         ['GeForce RTX 3070 Ti',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3070ti/'],
                         ['GeForce RTX 3070',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3070/'],
                         ['Radeon RX 6800',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6800/'],
                         ['GeForce RTX 3060 Ti',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3060ti/'],
                         ['Radeon RX 6750 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6750xt/'],
                         ['Radeon RX 6700 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6700xt/'],
                         ['Radeon RX 6700', #There are very few rx 6700 cards out there, and this is one of the cheapest.
                         'https://www.newegg.com/sapphire-radeon-rx-6700-11321-02-20g/p/N82E16814202424?Item=N82E16814202424&Description=rx%206700&cm_re=rx_6700-_-14-202-424-_-Product'],
                         ['GeForce RTX 3060',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3060/'],
                         ['GeForce RTX 2060 SUPER',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx2060super/'],
                         ['Radeon RX 6650 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6650xt/'],
                         ['Radeon RX 6600 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6600xt/'],
                         ['Radeon RX 6600',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6600/'],
                         ['GeForce RTX 2060',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx2060/'],
                         ['GeForce GTX 1660 SUPER',
                         'https://www.newegg.com/evga-geforce-gtx-1660-super-06g-p4-1068-kr/p/N82E16814487479?Item=9SIAN0MHXX7631&Description=1660%20super&cm_re=1660_super-_-14-487-479-_-Product&quicklink=true'],
                         ['GeForce RTX 3050',
                         'https://www.nowinstock.net/computers/videocards/nvidia/rtx3050/'],
                         ['GeForce GTX 1650 SUPER',
                         'https://www.newegg.com/evga-geforce-gtx-1650-super-04g-p4-1357-kr/p/N82E16814487482?Item=9SIANWTJ2E4196&Description=1650%20super&cm_re=1650_super-_-14-487-482-_-Product'],
                         ['Radeon RX 6500 XT',
                         'https://www.nowinstock.net/computers/videocards/amd/rx6500xt/']
                        ]   

In [12]:
for i in [0,1,2,3,4,5,6,7,8,9,10,11,12,14,15,16,17,18,19,21,23]:
    A = BeautifulSoup(requests.get(list_of_scores_to_fix[i][1]).text,'lxml')
    Ax=A.findAll('td',{'align':'right'})
    list_=[]
    for item in Ax:
        if 'data' in str(item):
            list_.append(item)
    list_2=[]
    
    bad_stuff=['0.00">S','1.00">$']
    for element in list_:
        list_2.append(str(element)[30:37])
    list_=[]
    for element in list_2:
        if element not in bad_stuff:
            list_.append(element)
    list_2=list_

    list_=[]
    for entry in list_2:
        a=str.replace(entry,'"','')
        list_.append(float(a))
    df.at[df[df['GPU']==list_of_scores_to_fix[i][0]].index.values[0],'NEW_PRICE']=min(list_)

In [13]:
# Newegg scraping...

A = BeautifulSoup(requests.get(list_of_scores_to_fix[13][1]).text,'lxml')
Ax=A.findAll('strong')
Ay=A.findAll('sup')

Ax=str.replace(str(Ax[1]),'<strong>','')
Ax=str.replace(str(Ax),'</strong>','')

Ay=str.replace(str(Ay),'[<sup>','')
Ay=str.replace(str(Ay),'</sup>]','')
min_=float(Ax+Ay)
df.at[df[df['GPU']==list_of_scores_to_fix[13][0]].index.values[0],'NEW_PRICE']=min_

In [14]:
# Newegg scraping...

A = BeautifulSoup(requests.get(list_of_scores_to_fix[20][1]).text,'lxml')
Ax=A.findAll('strong')
Ay=A.findAll('sup')

Ax=str.replace(str(Ax[2]),'<strong>','')
Ax=str.replace(str(Ax),'</strong>','')

Ay=str.replace(str(Ay[3]),'<sup>','')
Ay=str.replace(str(Ay),'</sup>','')
min_=float(Ax+Ay)
df.at[df[df['GPU']==list_of_scores_to_fix[20][0]].index.values[0],'NEW_PRICE']=min_

In [15]:
# Newegg scraping...

A = BeautifulSoup(requests.get(list_of_scores_to_fix[22][1]).text,'lxml')
Ax=A.findAll('strong')
Ay=A.findAll('sup')

Ax=str.replace(str(Ax[2]),'<strong>','')
Ax=str.replace(str(Ax),'</strong>','')

Ay=str.replace(str(Ay[3]),'<sup>','')
Ay=str.replace(str(Ay),'</sup>','')
min_=float(Ax+Ay)
df.at[df[df['GPU']==list_of_scores_to_fix[22][0]].index.values[0],'NEW_PRICE']=min_

In [16]:
df.at[128,'PASSMARK_SCORE']=np.nan

df['USED_DOLLARS_PER_POINT']=df['USED_PRICE']/df['PASSMARK_SCORE']
df['NEW_DOLLARS_PER_POINT']=df['NEW_PRICE']/df['PASSMARK_SCORE']

USED=df.sort_values('USED_DOLLARS_PER_POINT')
NEW=df.sort_values('NEW_DOLLARS_PER_POINT')
NEW['NEW_DOLLARS_PER_POINT']=NEW['NEW_PRICE']/NEW['PASSMARK_SCORE']
NEW.sort_values('NEW_DOLLARS_PER_POINT',inplace=True)
NEW=NEW.reset_index().drop('index',axis=1)
NEW

RANKING                                 GPU  NEW_PRICE  PASSMARK_SCORE  \
0         91              GeForce GTX 1660 SUPER     219.99         12762.0   
1         74                    GeForce RTX 2060     249.99         13988.0   
2        149                   Radeon RX 6500 XT     167.99          9259.0   
3         73                      Radeon RX 6600     259.99         14055.0   
4        131              GeForce GTX 1650 SUPER     196.99         10055.0   
5        104                    GeForce GTX 1660     229.99         11713.0   
6         16                 GeForce RTX 3060 Ti     399.99         20359.0   
7         29                      Radeon RX 6700     369.99         18726.0   
8        150                   Radeon RX 5500 XT     184.99          9191.0   
9         92                    GeForce RTX 3050     259.99         12749.0   
10        47              GeForce RTX 2060 SUPER     339.99         16508.0   
11        56                   Radeon RX 6600 XT     329.99         15863.0   
12        53                   Radeon RX 6650 XT     349.99         15934.0   
13       110                     GeForce GTX 980     247.35         11247.0   
14        38                    GeForce RTX 3060     379.99         17006.0   
15        11                    GeForce RTX 3070     499.99         22209.0   
16        27                   Radeon RX 6700 XT     448.99         18887.0   
17       139                     GeForce GTX 970     237.99          9673.0   
18       119                     Radeon RX 6600M     259.99         10529.0   
19        82                    GeForce GTX 1070     342.55         13494.0   
20        10                 GeForce RTX 3070 Ti     599.99         23572.0   
21        24                   Radeon RX 6750 XT     499.99         19205.0   
22        33                 GeForce GTX 1080 Ti     483.82         18361.0   
23        70                 GeForce GTX 1070 Ti     389.00         14577.0   
24        35              GeForce RTX 2070 SUPER     499.99         18188.0   
25       136                GeForce GTX 1060 3GB     269.99          9733.0   
26        15                      Radeon RX 6800     579.99         20868.0   
27       130                    GeForce GTX 1060     279.99         10068.0   
28         6                    GeForce RTX 3080     699.00         25132.0   
29       102                 GeForce GTX 1660 Ti     329.99         11808.0   
30         8                   Radeon RX 6800 XT     669.99         23793.0   
31        60                    GeForce GTX 1080     442.79         15298.0   
32         5                   Radeon RX 6900 XT     749.99         25757.0   
33        90         GeForce RTX 3060 Laptop GPU     384.99         12827.0   
34         3                 GeForce RTX 3080 Ti     899.99         27207.0   
35       156         GeForce RTX 3050 Laptop GPU     300.02          9039.0   
36        76                  GeForce GTX 980 Ti     465.66         13869.0   
37        28                    GeForce RTX 2080     633.03         18727.0   
38        21              GeForce RTX 2080 SUPER     683.99         19509.0   
39         2                   Radeon RX 6950 XT     999.99         27987.0   
40       146                  GeForce GTX 780 Ti     339.99          9364.0   
41        50                    GeForce RTX 2070     627.44         16083.0   
42        78                   Radeon RX 5600 XT     545.00         13846.0   
43       158                       Radeon RX 580     364.01          8918.0   
44        41                   Radeon RX 5700 XT     694.94         16899.0   
45        80                      NVIDIA TITAN X     608.10         13660.0   
46         4                    GeForce RTX 3090    1203.18         26560.0   
47        12                 GeForce RTX 2080 Ti     998.59         21817.0   
48       135             Radeon R9 Fury + Fury X     449.99          9741.0   
49       141                      Radeon R9 Fury     44

In [17]:
USED=USED.reset_index().drop('index',axis=1)
USED

RANKING                                 GPU  NEW_PRICE  PASSMARK_SCORE  \
0        139                     GeForce GTX 970     237.99          9673.0   
1        155                       Radeon R9 390        NaN          9044.0   
2        146                  GeForce GTX 780 Ti     339.99          9364.0   
3        136                GeForce GTX 1060 3GB     269.99          9733.0   
4        110                     GeForce GTX 980     247.35         11247.0   
5         76                  GeForce GTX 980 Ti     465.66         13869.0   
6        104                    GeForce GTX 1660     229.99         11713.0   
7         82                    GeForce GTX 1070     342.55         13494.0   
8        141                      Radeon R9 Fury     443.10          9564.0   
9         70                 GeForce GTX 1070 Ti     389.00         14577.0   
10       130                    GeForce GTX 1060     279.99         10068.0   
11        60                    GeForce GTX 1080     442.79         15298.0   
12        91              GeForce GTX 1660 SUPER     219.99         12762.0   
13       158                       Radeon RX 580     364.01          8918.0   
14        33                 GeForce GTX 1080 Ti     483.82         18361.0   
15        78                   Radeon RX 5600 XT     545.00         13846.0   
16       131              GeForce GTX 1650 SUPER     196.99         10055.0   
17        74                    GeForce RTX 2060     249.99         13988.0   
18        68                   Radeon RX Vega 64     799.99         14661.0   
19       127                    Radeon R9 Fury X        NaN         10142.0   
20        80                      NVIDIA TITAN X     608.10         13660.0   
21       150                   Radeon RX 5500 XT     184.99          9191.0   
22        83                 GeForce GTX TITAN X    1096.99         13279.0   
23       102                 GeForce GTX 1660 Ti     329.99         11808.0   
24       140                       Radeon RX 590     549.99          9664.0   
25        47              GeForce RTX 2060 SUPER     339.99         16508.0   
26       142                      Radeon R9 390X        NaN          9476.0   
27        73                      Radeon RX 6600     259.99         14055.0   
28        81                   Radeon RX Vega 56     752.99         13639.0   
29        35              GeForce RTX 2070 SUPER     499.99         18188.0   
30        21              GeForce RTX 2080 SUPER     683.99         19509.0   
31        50                    GeForce RTX 2070     627.44         16083.0   
32        56                   Radeon RX 6600 XT     329.99         15863.0   
33        41                   Radeon RX 5700 XT     694.94         16899.0   
34        28                    GeForce RTX 2080     633.03         18727.0   
35        67                      Radeon RX 5700     899.95         14668.0   
36       113                      Radeon RX 5600        NaN         11177.0   
37        22                            TITAN Xp        NaN         19327.0   
38        38                    GeForce RTX 3060     379.99         17006.0   
39        12                 GeForce RTX 2080 Ti     998.59         21817.0   
40        11                    GeForce RTX 3070     499.99         22209.0   
41        27                   Radeon RX 6700 XT     448.99         18887.0   
42        16                 GeForce RTX 3060 Ti     399.99         20359.0   
43        55                    RadeonT RX 6850M        NaN         15899.0   
44        10                 GeForce RTX 3070 Ti     599.99         23572.0   
45        92                    GeForce RTX 3050     259.99         12749.0   
46        15                      Radeon RX 6800     579.99         20868.0   
47         8                   Radeon RX 6800 XT     669.99         23793.0   
48         6                    GeForce RTX 3080     699.00         25132.0   
49        24                   Radeon RX 6750 XT     49

In [18]:
NEW.to_csv('New GPU Price to Performance.csv')
USED.to_csv('Used GPU Price to Performance.csv')

## There isn't an absolute conclusion to make here. It depends on what tier of performance you want. For instance, the best NEW card above 20,000 points in Passmark is the 3060 ti, so if you need a relatively high performance card for the cheapest you can get it, that's one possibility.

## If you're just looking for the max price to performance you can get your hands on regardless of actual relative performance, the GTX 970 still holds up (this happens to be the GPU I currently have).

## If you want the best performance period, then the first iteration of the dataframe DF shows that the 3090 ti will be your best friend.

## If you don't mind buying used, but have a limit of 250 dollars maximum, you can't get much better than the 1080 ti.

## And obviously, since I have this code scraping code directly from the websites, these are all subject to change.
